.github/workflows/GPU_Beltrami_build_run.yml

```diff
@@ -0,0 +1,22 @@
+name: Samurai GPU test with Beltrami
+
+on:
+  workflow_dispatch:
+    branches: [ "main" ]
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+
+    runs-on: [self-hosted, Linux, X64]
```

Comment:
@sjsprecious As i mentioned in the slack yes the CPU workflow is going over the resource limit for normal CI workflow. This is mainly because of including the whole samurai dataset. We need to discuss different approach of getting the data. 
Regarding the GPU workflow, as far as I know github doesn't provide GPU resources (even if they provide its probably gonna be on paid their). The GPU CI jobs will keep waiting for the resources to be allocated otherwise it will timeout after a certain time.

Reply:
is it possible to get smaller dataset?